### DISCUSSION TITLE
Too few arguments to function Spatie\Permission\Models\Role::scopePermission()

### DISCUSSION THREAD

**Discussion by horsford92** [spatie/laravel-permission#2144] [Q&A] [answered]
Good morning... when executing migrate --seed in the terminal it throws me this error... I don't know what to do anymore... thanks in advance for the help.. Iam using Laravel 9

ArgumentCountError 

  Too few arguments to function Spatie\Permission\Models\Role::scopePermission(), 1 passed in C:\xampp\htdocs\SGI\vendor\laravel\framework\src\Illuminate\Database\Eloquent\Model.php on line 1507 and exactly 2 expected

  at C:\xampp\htdocs\SGI\vendor\spatie\laravel-permission\src\Traits\HasPermissions.php:71
     67▕      * @param string|int|array|\Spatie\Permission\Contracts\Permission|\Illuminate\Support\Collection $permissions
     68▕      *
     69▕      * @return \Illuminate\Database\Eloquent\Builder
     70▕      */
  ➜  71▕     public function scopePermission(Builder $query, $permissions): Builder
     72▕     {
     73▕         $permissions = $this->convertToPermissionModels($permissions);
     74▕
     75▕         $rolesWithPermissions = array_unique(array_reduce($permissions, function ($result, $permission) {

  1   C:\xampp\htdocs\SGI\vendor\laravel\framework\src\Illuminate\Database\Eloquent\Model.php:1507
      Spatie\Permission\Models\Role::scopePermission(Object(Illuminate\Database\Eloquent\Builder))

  2   C:\xampp\htdocs\SGI\vendor\laravel\framework\src\Illuminate\Database\Eloquent\Builder.php:1296
      Illuminate\Database\Eloquent\Model::callNamedScope("permission")

**Comment by horsford92** [comment]
@erikn69

**Comment by erikn69** [comment, marked as answer]
`migrate --seed` this package don't use that, so, you write something wrong, impossible to know what, but it is working for everyone
If you want more help make a Example App on a minimal application in a clean installation where you demonstrate the problem, surely when you do that you will find what you configured wrong, or someone will help you to find it
https://spatie.be/docs/laravel-permission/v5/basic-usage/new-app#content-initial-setup

**Reply by horsford92** [reply]
sorry I didn't put the full `php artisan migrate:fresh --seed `command, this is the `RoleSeeder.php` file

```
class RoleSeeder extends Seeder
{
    /**
     * Run the database seeds.
     *
     * @return void
     */
    public function run()
    {
        $roleadmin = Role::create(['name' => 'Admin']);
        $rolemoderador = Role::create(['name' => 'Especialista']);


        //Permisos dashboard
        Permission::create(['name' => 'dash.index']) -> syncRoles([$roleadmin, $rolemoderador]);


        //Permisos Gestionar Usuarios
        Permission::create(['name' => 'users.index']) -> assignRole($roleadmin);
        Permission::create(['name' => 'users.create']) -> assignRole($roleadmin); 
        Permission::create(['name' => 'users.show']) -> assignRole($roleadmin);
        Permission::create(['name' => 'users.destroy']) -> assignRole($roleadmin);

        $roleadmin->permission()->attach();
    }
}
```

and then running the command `php artisan migrate:fresh --seed` throws me the following error

```
ArgumentCountError 

  Too few arguments to function Spatie\Permission\Models\Role::scopePermission(), 1 passed in C:\xampp\htdocs\SGI\vendor\laravel\framework\src\Illuminate\Database\Eloquent\Model.php on line 1507 and exactly 2 expected

  at C:\xampp\htdocs\SGI\vendor\spatie\laravel-permission\src\Traits\HasPermissions.php:71
     67▕      * @param string|int|array|\Spatie\Permission\Contracts\Permission|\Illuminate\Support\Collection $permissions
     68▕      *
     69▕      * @return \Illuminate\Database\Eloquent\Builder
     70▕      */
  ➜  71▕     public function scopePermission(Builder $query, $permissions): Builder
     72▕     {
     73▕         $permissions = $this->convertToPermissionModels($permissions);
     74▕
     75▕         $rolesWithPermissions = array_unique(array_reduce($permissions, function ($result, $permission) {

  1   C:\xampp\htdocs\SGI\vendor\laravel\framework\src\Illuminate\Database\Eloquent\Model.php:1507
      Spatie\Permission\Models\Role::scopePermission(Object(Illuminate\Database\Eloquent\Builder))

  2   C:\xampp\htdocs\SGI\vendor\laravel\framework\src\Illuminate\Database\Eloquent\Builder.php:1296
      Illuminate\Database\Eloquent\Model::callNamedScope("permission")
```

**Reply by erikn69** [reply]
> $roleadmin->permission()->attach();
